### ISSUE TITLE
Create DropdownOption for slack integration

### ISSUE THREAD

**Issue by zhx828** [oncokb/oncokb#3527]
We will have a DropdownEmailOption be included in the https://github.com/oncokb/oncokb-public/pull/981. But ideally we should create a generic DropdownOption by specifying category, action id, email. Not all options are related to sending email. But they all should belong to a category and have a action id associated.

**Comment by benzuzu** [comment]
Good idea. I did consider this, but my one reservation was that all the DropdownEmailOption's have the same logic. Any non email options would definitely have different logic from the email options (and each other), so I figured it would be a bit confusing, since the point of the abstraction was to encapsulate the general process into a single class. However, it does make more sense for all the dropdown options to be grouped together.
